Source/missiles.cpp

```diff
@@ -3628,19 +3616,18 @@ void MI_Firemove(MissileStruct &missile)
 		AddUnLight(missile._mlid);
 	}
 	if (missile._mimfnum != 0 || missile._mirange == 0) {
-		if (missile.position.tile.x != missile.var3 || missile.position.tile.y != missile.var4) {
+		if (missile.position.tile != Point { missile.var3, missile.var4 }) {
 			missile.var3 = missile.position.tile.x;
 			missile.var4 = missile.position.tile.y;
-			ChangeLight(missile._mlid, { missile.var3, missile.var4 }, 8);
+			ChangeLight(missile._mlid, missile.position.tile, 8);
 		}
 	} else {
 		if (missile.var2 == 0)
 			missile._mlid = AddLight(missile.position.tile, ExpLight[0]);
 		ChangeLight(missile._mlid, missile.position.tile, ExpLight[missile.var2]);
 		missile.var2++;
 	}
-	missile.position.tile.x++;
-	missile.position.tile.y++;
+	missile.position.tile += Displacement { 1, 1 };

```

Comment:
```suggestion
	missile.position.tile += DIR_S;
```
No idea why this missile is being shifted south one tile... This is being done for other missiles too, seems to always be a shift north or south.